src/config.ts

```diff
@@ -682,21 +692,45 @@ function setDeepProperty(obj, value, target) {
     }
 }
 
+export function getURL(url, target) {
+    let url_match =
+        // For each key
+        Object.keys(URLCONFIGS)
+            // Create a tuple containing the key and its match
+            .map(k => [k, url.match(k)])
+            // Keep only the ones that have a match
+            .filter(([_, m]) => m)
+            // Sort them from the longest match to the shortest
+            .sort(([k1, m1], [k2, m2]) => m2.length - m1.length)
+            // Get the first config name that has `target`
+            .find(([k, _]) => getDeepProperty(URLCONFIGS[k], target))
+
+    if (url_match && url_match[0])
+        return getDeepProperty(URLCONFIGS[url_match[0]], target)
+}
+
 /** Get the value of the key target.
 
     If the user has not specified a key, use the corresponding key from
     defaults, if one exists, else undefined.
     @hidden
 */
 export function get(...target) {
+    // Window.tri might not be defined when called from the untrusted page context
+    let loc = window.location
+    if ((window as any).tri) loc = (window as any).tri.contentLocation
```

Comment:
Can we use contentLocation as the fallback rather than the default? I'm... kind of not a fan of the contentLocation idea. Even if 2ms is enough for you, some users are on faster computers and tab-switching happens enough that I'll predict more breakage than less. I'd also expect it to break continually when tabs do things when they're not the active tab, for example, if they navigate on their own, or if someone's performance.now() is less accurate than 2ms for any reason.

Reply:
`contentLocation` is always `window.location` in the content script (see [this](https://github.com/tridactyl/tridactyl/pull/999/files#diff-a33a69a9530b9842007c0dad7c7f1dc5R71)), it only becomes this weird changing value in the background script. I'm not sure how you would make it a fallback since `contentLocation` is the only value we can use while in the background script.

I agree that my race-condition protection code isn't great, but I have no idea how else we could implement it. You can indeed expect it to break when tabs do things while not being the active tab, but only if they message the background process, and this already breaks pretty hard since we use `activeTabId()` everywhere.

Reply:
> contentLocation is always window.location in the content script (see this), it only becomes this weird changing value in the background script. I'm not sure how you would make it a fallback since contentLocation is the only value we can use while in the background script.

Ah, I'd missed that it was set differently in the content script. >_>

Are there any ways for the location to change without the content script being reloaded? SPAs mucking around with the stuff after the hash, anything like that?

> You can indeed expect breakage when tabs do things while not being the active tab, but only if they message the background process, and this already breaks pretty hard since we use activeTabId() everywhere.

I guess. I'm still trying to fix that, but since it's already involving huge changes across the entire project I suppose a few more won't change much.

Reply:
I think you're right, I'll remove the contentLocation stuff. It's better to have the background process always return the global/default setting rather than one from a completely unrelated page.